core/src/main/scala/scalaz/LazyEitherT.scala

```diff
@@ -105,7 +105,7 @@ object LazyEitherT extends LazyEitherTInstances {
     u1: Unapply[Functor, FAB]{type A = AB},
     u2: Unapply2[Bifunctor, AB]{type A = A0; type B = B0},
     l: Leibniz.===[AB, LazyEither[A0, B0]]
-  ): LazyEitherT[u1.M, A0, B0] = LazyEitherT(l.subst[u1.M](u1(fab)))
+  ): LazyEitherT[u1.M, u2.A, u2.B] = LazyEitherT(l.subst[u1.M](u1(fab)))
```

Comment:
That's also why 1110a5e94fbee1f0d324fb70722d9a4dcd7aa008 didn't include an `Aux` variant including the hk member; scala/bug#5075 makes it so it never works.